mender-update/update_module/v3/platform/boost_filesystem/fs_operations.cpp

```diff
@@ -207,6 +218,31 @@ error::Error UpdateModule::PrepareFileTree(
 	return SyncFs(path);
 }
 
+error::Error UpdateModule::EnsureRootfsImageFileTree(const string &path) {
+	// Historical note: Versions of the client prior to 4.0 had the rootfs-image module built
+	// in. Because of this it has no Update Module File Tree. So if we are upgrading, we might
+	// hit an on-going upgrade without a File Tree. It's too late to create a complete one with
+	// all the artifact content by the time we get here, but at least we can create one which
+	// has the current Provides information, as well as a folder for the Update Module to run
+	// in.
+	ifstream payload_type(path::Join(path, "header", "payload_type"));
```

Comment:
It doesn't help in this particular case because we are not handling the error, we are just checking for the existence of any error at all. So the original is actually shorter, since we don't need `expected` handling.